src/core/bitop.d

```diff
@@ -706,3 +706,38 @@ version (D_InlineAsm_X86_64)
         }
     }
 }
+
+/**
+ *  Bitwise rotates unsigned arithmetic types
+ */
+pure T rotl(T)(in T u, in uint s)
+    if (__traits(isArithmetic, T) && __traits(isUnsigned, T))
+{
+    return cast(T) ((u << s) | (u >> (-s & (T.sizeof * 8 - 1))));
```

Comment:
What happens when `s` is bigger than `8 * T.sizeof`? If that behavior is inconsistent across architectures, an `assert(s < 8 * T.sizeof)` would be adequate.